efr32mg_sltb004a: Build issue on `tests/drivers/spi/spi_loopback/drivers.spi.loopback`

**Describe the bug**
CI is failing on this board due to the following error:
```
/__w/zephyr/zephyr/drivers/spi/spi_gecko.c
/__w/zephyr/zephyr/drivers/spi/spi_gecko.c: In function 'spi_gecko_xfer':
/__w/zephyr/zephyr/drivers/spi/spi_gecko.c:176:25: error: redefinition of 'data'
  176 |  struct spi_gecko_data *data = dev->data;
      |                         ^~~~
/__w/zephyr/zephyr/drivers/spi/spi_gecko.c:173:25: note: previous definition of 'data' was here
  173 |  struct spi_gecko_data *data = dev->data;
      |                         ^~~~
```

**Impact**
Blocks CI, cf https://github.com/zephyrproject-rtos/zephyr/runs/4934340081?check_suite_focus=true

--- a/drivers/spi/spi_gecko.c
+++ b/drivers/spi/spi_gecko.c
@@ -173,7 +173,6 @@ static void spi_gecko_xfer(const struct device *dev,
 	struct spi_gecko_data *data = dev->data;
 	struct spi_context *ctx = &data->ctx;
 	const struct spi_gecko_config *gecko_config = dev->config;
-	struct spi_gecko_data *data = dev->data;
 
 	spi_context_cs_control(ctx, true);
 

